### DISCUSSION TITLE
i need put two connection string to deploy in local and other in server  in file  App_Data/Sites/appsettings.json

### DISCUSSION THREAD

**Discussion by ODA669** [OrchardCMS/OrchardCore#14526] [General]
good morning , someone know how i can add two connection string un appsettings.json to App_Date/Sites to project orchard core i need insert ones connection string to work in local and other for deploy them i have this
 {
"DatabaseProvider": "SqlConnection",
"ConnectionString": ""
"Default": {
  "State": "Uninitialized",
  "TablePrefix": ""
}

 i try to do this but don't work
 
"DatabaseProvider": "SqlConnection",
"ConnectionStrings": {
    "ConnectionString1": "TuConnectionString1",
    "ConnectionString2": "TuConnectionString2"
  },
"Default": {
  "State": "Uninitialized",
  "TablePrefix": ""
}{

**Comment by jtkech** [comment]
We can use different config sources, there are 2 default ones, one being the main appsettings of the app, and we have for each tenant a specific appsettings in the `App_Data/Sites/{theTenant}/` folder. Note: We also have a `tenants.json` directly under  `App_Data` but for internal usage. But we don't parse any appsettings directly under `App_Data/Sites`.

If it is before any setup you can use the main appsettings, under the section `OrchardCore`, if you only have one tenant (so the `Default` one) you don't need a subsection matching the tenant name, then just define the config values you want.

If you want different configurations based on the environment you can use `appsettings[.env].json`, for example `appsettings.json` and `appsettings.Development.json`, if in `Development` a config in `appsettings.json` may still be used unless if also defined (maybe not all) in `appsettings.Development.json` with a different value.

Note: You can also have an `appsettings.Production.json` for example or any other custom environment names.

---

If it is after a setup where these configs were not provided in a previous source, they will be persisted in the tenant specific folder, here we don't support any `appsettings[.env].json` based on the environment.

What you can do is to remove these configs from the tenant specific appsettings and then add them in some main appsettings based on the environment as described above.

Notice that in the tenant specific appsettings we neither need a tenant section nor the `OrchardCore` section, but the `OrchardCore` section is required in a main appsettings.

Note: The `"State":` field is not to configure an existing tenant but to pre-create and pre-configure a tenant, so when it doesn't exist yet we need to make the distinction between a tenant name and any other config key.

**Comment by ODA669** [comment]
thanks
